contracts/Content/UniqueContent.sol

```diff
@@ -0,0 +1,173 @@
+        IERC1155Upgradeable(_data.contentAddress).safeTransferFrom(_data.to, address(this), _data.tokenId, 1, "");
+        _mint(_data.to, uniqueIdsCounter);
+        
+        uniqueAssetInfo[uniqueIdsCounter].creator = _msgSender();
+        uniqueAssetInfo[uniqueIdsCounter].contentAddress = _data.contentAddress;
+        uniqueAssetInfo[uniqueIdsCounter].tokenId = _data.tokenId;
+        uniqueAssetInfo[uniqueIdsCounter].uniqueAssetUri.push(_data.uniqueAssetUri);
+        uniqueAssetInfo[uniqueIdsCounter].creatorLocked = _data.creatorLocked;
+        _setTokenRoyalties(uniqueIdsCounter, _data.royaltyReceivers, _data.royaltyRates);
+        
+        emit UniqueUriUpdated(uniqueIdsCounter, 0);
+        emit Mint(_msgSender(), _data);
+
+        // increment uniqueIdsCounter for next minting process
+        uniqueIdsCounter++;
+    }
+
+    /** Asset Burning
+    * @dev If the caller is the owner of the token and if the unique asset is not creator locked, it returns the original asset, burns the unique asset, and then deletes the unique asset's token info
+    * @param _uniqueId uint256 ID of token to burn
+    */
+    function burn(uint256 _uniqueId) external override {
+        require(ownerOf(_uniqueId) == _msgSender(), "You must be the owner of this token to burn it");
+        require(!uniqueAssetInfo[_uniqueId].creatorLocked, "Creator has disabled the burning of this token");
+        // transfers original asset back to caller
+        IERC1155Upgradeable(uniqueAssetInfo[_uniqueId].contentAddress).safeTransferFrom(address(this), _msgSender(), uniqueAssetInfo[_uniqueId].tokenId, 1, "");
+        _burn(_uniqueId);
+
+        uniqueAssetInfo[_uniqueId].creator = address(0);
+        uniqueAssetInfo[_uniqueId].contentAddress = address(0);
+        uniqueAssetInfo[_uniqueId].tokenId = 0;
+        uniqueAssetInfo[_uniqueId].version = 0;
+        uniqueAssetInfo[_uniqueId].creatorLocked = false;
+        delete uniqueAssetInfo[_uniqueId].uniqueAssetUri;
+        delete tokenRoyalties[_uniqueId];
+        
+        emit Burn(_msgSender(), _uniqueId);
+    }
+
+    /**
+    * @dev Returns the uri of a specific version of the original asset the unique asset is based on
+    * @param _uniqueId uint256 ID of token to query original asset uri of
+    * @param _version version number of token to query
+    */
+    function originalAssetUri(uint256 _uniqueId, uint256 _version) external view override returns (string memory) {
+        require(_exists(_uniqueId), "Unique Id does not exist");
+        uint256 _tokenId = uniqueAssetInfo[_uniqueId].tokenId;
+        return IContent(uniqueAssetInfo[_uniqueId].contentAddress).uri(_tokenId, _version);
+    }
+
+    /**
+    * @dev Returns the unique asset uri of a specific version
+    * @param _uniqueId uint256 ID of token to query
+    * @param _version version number of token to query
+    */
+    function uri(uint256 _uniqueId, uint256 _version) external view override returns (string memory) {
```

Comment:
update this to tokenURI()

Reply:
you also need to override tokenUri(uint256 tokenId) from ERC721, because that doesn't return uniqueAssetInfo[uniqueId].uniqueAssetUri[currentVersion]